src/dashboard/src/main/models.py

```diff
@@ -871,6 +877,26 @@ class Meta:
         verbose_name = _('Rights: Agent')
 
 
+class UnitVariableManager(models.Manager):
+
+    def set_variable(self, unit_type, unit_uuid,
+                     variable, value, link_id=None):
+        """Update or create unit variable."""
+        defaults = {
+            "variablevalue": value,
+            "microservicechainlink": link_id,
+        }
+        return self.get_queryset().update_or_create(
+            unittype=unit_type, unituuid=unit_uuid,
+            variable=variable, defaults=defaults)
+
+    def set_active_agent(self, unit_type, unit_id, user_id):
+        """Set active agent given the user ID."""
+        return self.set_variable(
+            unit_type, unit_id,
+            "activeAgent", User.objects.get(id=user_id).userprofile.agent.id)
+
```

Comment:
There's probably more and better ways to do this.
I'm definitely not used to work with Django models.

Reply:
I don't actually think there is really...but maybe I'm misunderstanding things. I like that you're using a manager here, that's the correct place for logic that operates on this level (across multiple model instances). I have a few more nitpicky suggestions: 
* I would break out `User.objects.get(id=user_id).userprofile.agent.id` as above
* I guess you can make the user query more efficient with `select_related` and just accessing the agent as `userprofile.agent_id` (something like `User.objects.select_related('userprofile').get(id=user_id).userprofile.agent_id` which I haven't tested 😄 )
* I think as a general rule it's better to use `update_foo` or `save_foo` or something vs `set_foo` when dealing with the ORM — makes it clearer that calling the method hits the DB and it's not just a setter/getter. So minor though.

Reply:
Thank you! Nice to learn about `select_related`!